### PR TITLE
Braintree java issue 119

### PR DESCRIPTION
**Pull request by kmichalak**
# Summary

# Checklist

- [x] Added changelog entry
- [x] Ran unit tests (mvn verify -DskipITs)
- [x] I understand that unless this is a Draft PR or has a DO NOT MERGE label, this PR is considered to be in a deploy ready state and can be deployed if merged to main

### CODE REVIEW

**Comment by hollabaq86** [comment]
Thanks @kmichalak! We;ll take a look and provide feedback is needed. 

For internal tracking, issue 7173

File: src/main/java/com/braintreegateway/ClientTokenGateway.java [hunk 1]
```diff
@@ -32,15 +34,18 @@ public String generate() {
     }
 
     public String generate(ClientTokenRequest request) {
-      NodeWrapper response = null;
+
```
Comment (saralvasquez) [inline src/main/java/com/braintreegateway/ClientTokenGateway.java:37]: Can you delete this empty line?

File: src/test/java/com/braintreegateway/unittest/ClientTokenGatewayTest.java [hunk 2]
```diff
@@ -60,4 +74,52 @@ public void generateRaisesExceptionIfFailOnDuplicatePaymentMethodIsIncludedWitho
         assertTrue(e.getMessage().contains("failOnDuplicatePaymentMethod"),
                 "exception message should mention failOnDuplicatePaymentMethod");
     }
+
+    @Test
+    void generateRaisesExceptionIfApiReturnsError() {
```
Comment (saralvasquez) [inline src/test/java/com/braintreegateway/unittest/ClientTokenGatewayTest.java:79]:
```suggestion
    public void generateRaisesExceptionIfApiReturnsError() {
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## 3.21.1
```
Comment (saralvasquez) [inline CHANGELOG.md:3]:
```suggestion
## unreleased
```

**Review by saralvasquez — COMMENTED** [review]
This looks really great @kmichalak. Thank you for putting this together! Just a couple nits

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
...
+* Fix issue where null is returned as a client token value when customer matching customer_id from the request does not exist.
```
[the remark this answers is not in the document]
Comment (kmichalak) [reply CHANGELOG.md]: Thanks for suggestions. I can see that you've fixed this already :)

**Comment by saralvasquez** [comment]
@kmichalak thanks for that update! We'll give it a final pass and get it pulled in

**Comment by saralvasquez** [comment]
Hi! Sorry this process has taken so long. My team and I have been discussing this issue for a while since it's a lot more complicated than it initially seemed. Ultimately, we concluded that we can't pull in these changes. I will post a detailed explanation in the [open GitHub issue](https://github.com/braintree/braintree_java/issues/119), but the TLDR of it is that these changes would break user integrations by changing the base functionality of the `generate` method. We have to assume users aren't set up to automatically handle new exceptions thrown when the generate call would normally just return null. In order to stay in line with semantic versioning, we can't break users' integrations unless we put out a new major version, which we don't have the bandwidth for at the moment. We do plan on changing how client token generation handles nonexistent customers in all of the SDKs and we will find a way to credit you in the Java SDK when the next major version comes out. Thank you so much for putting in this work! We're really sorry it won't be used just yet.

**Pull request closed without merging**